### PR TITLE
fix(quick-add): disable add work-item for non-collaborator

### PR DESCRIPTION
**Pull request by divyanshiGupta**
<strong>Note: If there are pending changes to the PR, prefix the PR title with "WIP" and add the label "DO NOT MERGE"</strong>

### Mandatory
 - [ ] What does this PR do? 
        
It disables add work-item for non-collaborators except for in Openshift_io space as currently we allow non-collaborators to add work-items in Openshift_io space and since the permissions model is not yet ready so this is a temporary fix.
- [ ] What issue/task does this PR references?
       https://github.com/openshiftio/openshift.io/issues/3615

- [ ] Are the tests Included?
___
### Optional
- [ ] Is the documentation Included?

- [ ] Are the Release Notes included?

- [ ] @mention(s) to expected reviewer(s) included

### CODE REVIEW

**Comment by divyanshiGupta** [comment]
[test]

**Comment by divyanshiGupta** [comment]
[test]

**Comment by sahil143** [comment]
[test]

File: src/app/components_ngrx/work-item-quick-add/work-item-quick-add.component.ts [hunk 1]
```diff
@@ -86,7 +93,32 @@ export class WorkItemQuickAddComponent implements OnInit, OnDestroy, AfterViewIn
         .subscribe(items => {
           // const addedItem = items.find(item => item.createId === this.createId);
           this.resetQuickAdd();
-        })
+        }),
+      // This is temporary as the permissions model is not yet ready, currently the
+      // non-collaborator will only be able to add workitems in Openshift_io space
+      this.spaceQuery.getCurrentSpace
```
Comment (sanbornsen) [inline src/app/components_ngrx/work-item-quick-add/work-item-quick-add.component.ts]: Can we combine these two queries? Can we create a new query for this permission? This doesn't look like the best solution.

File: src/app/components_ngrx/work-item-quick-add/work-item-quick-add.component.ts [hunk 2]
```diff
@@ -86,7 +93,32 @@ export class WorkItemQuickAddComponent implements OnInit, OnDestroy, AfterViewIn
...
+        .subscribe(currentSpace => {
+          if (currentSpace.attributes.name == 'Openshift_io') {
+            this.ifOpenshift_io = true;
+          } else {
+          this.ifOpenshift_io = false;
+          }
+        }
+      ),
+      this.userQuery.getLoggedInUser.subscribe(currentUser => this.currentUserID = currentUser.id),
```
[the remark this answers is not in the document]
Comment (divyanshiGupta) [reply src/app/components_ngrx/work-item-quick-add/work-item-quick-add.component.ts]: @sahil143 it is getting unsubscribed in `ngOnDestroy()` as I have added these observables in the `eventListeners` array

[the remark this answers is not in the document]
Comment (sahil143) [reply src/app/components_ngrx/work-item-quick-add/work-item-quick-add.component.ts]: Ah! okay. I missed that.

**Comment by divyanshiGupta** [comment]
@sanbornsen can you take a look at this PR?

**Comment by divyanshiGupta** [comment]
@sanbornsen can you approve the PR ?

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
@divyanshiGupta I added me as a collaborator in space "osio-ci-planner-002-preview-SDD-space-2018-10-04T06-09-39" and tried adding a work item it works but when I removed as a collaborator .
Input box is enabled again

**Comment by Raunak1203** [comment]
> @divyanshiGupta I added me as a collaborator in space "osio-ci-planner-002-preview-SDD-space-2018-10-04T06-09-39" and tried adding a work item it works but when I removed as a collaborator .
> Input box is enabled again

It works after refresh

**Comment by divyanshiGupta** [comment]
@Raunak1203 it works for me, you need to refresh after removing yourself as a collaborator.

**Comment by Raunak1203** [comment]
@divyanshiGupta please disable onhover "Please fill out these field" instead of this can we add some on hover text which could give a hint to user why it is disabled ?

**Pull request merged**